agent/lib/kontena/workers/weave_worker.rb

```diff
@@ -45,22 +45,31 @@ def on_dns_add(topic, event)
       add_dns(event[:id], event[:ip], event[:name])
     end
 
+    # Was container event from the weave router?
+    def router_image?(image)
+      image.to_s == "#{Kontena::NetworkAdapters::Weave::WEAVE_IMAGE}:#{Kontena::NetworkAdapters::Weave::WEAVE_VERSION}"
```

Comment:
Moved those bits into the `WeaveHelper` module. Didn't touch the `WeaveExecutor`.